File: 프로그래머스/lv0/120882. 등수 매기기/등수 매기기.java
import java.util.*;

class Solution {
    public int[] solution(int[][] score) {
        ArrayList<Double> a = new ArrayList<>();
        int[] answer = new int[score.length];
        for (int i = 0; i < score.length; i++) {
            a.add(Math.ceil((score[i][0] + score[i][1]) /(double) 2));
        }
        Collections.sort(a,Comparator.reverseOrder());
        for (int i = 0; i < score.length; i++) {
            for (int j = 0; j < a.size(); j++) {
                if (Math.ceil((score[i][0] + score[i][1]) / (double)2) == a.get(j)) {
                    answer[i] = a.indexOf(a.get(j))+1;
                }
            }
        }
        return answer;
    }
}

File: 프로그래머스/unrated/120882. 등수 매기기/등수 매기기.js
function solution(score) {
  const averages = [];
  
  for (const s of score) {
    const average = (s[0] + s[1]) / 2;
    averages.push(average);
  }
  
  const rankings = Array.from({ length: averages.length }, () => 1);
  for (let i = 0; i < averages.length; i++) {
    for (let j = 0; j < averages.length; j++) {
      if (averages[i] < averages[j]) {
        rankings[i]++;
      }
    }
  }
  
  return rankings;
}